common/app/views/support/HtmlCleaner.scala

```diff
@@ -811,13 +811,17 @@ case class AffiliateLinksCleaner(pageUrl: String, sectionId: String, showAffilia
 }
 
 object AffiliateLinksCleaner {
-  def replaceLinksInHtml(html: Document, pageUrl: String, appendDisclaimer: Boolean, contentType: String, skimlinksId: String): Document = {
-    val links = html.getElementsByAttribute("href")
 
-    val supportedLinks: mutable.Seq[Element] = links.asScala.filter(isAffiliatable)
+  def getAffiliateableLinks(html:Document): mutable.Seq[Element] =
+    html.getElementsByAttribute("href").asScala.filter(isAffiliatable)
+
+  def replaceLinksInHtml(html: Document, pageUrl: String, appendDisclaimer: Option[Boolean], contentType: String, skimlinksId: String): Document = {
+
+    val supportedLinks: mutable.Seq[Element] = getAffiliateableLinks(html)
     supportedLinks.foreach{el => el.attr("href", linkToSkimLink(el.attr("href"), pageUrl, skimlinksId))}
 
-    if (supportedLinks.nonEmpty) insertAffiliateDisclaimer(html, contentType)
+    val shouldAppendDisclaimer = appendDisclaimer.getOrElse(supportedLinks.nonEmpty)
+    if (shouldAppendDisclaimer) insertAffiliateDisclaimer(html, contentType)
```

Comment:
if None then we assume it is last and rely on supportedLinks.nonEmpty?

Reply:
This logic isn't very clear - I've updated it with a comment to explain what's going on.